subsys/bluetooth/host/iso.c

```diff
@@ -1441,6 +1480,92 @@ int bt_iso_cig_terminate(struct bt_iso_cig *cig)
 	return 0;
 }
 
+void bt_iso_security_changed(struct bt_conn *acl, uint8_t hci_status)
+{
+	struct bt_iso_connect_param param[CONFIG_BT_MAX_CONN];
+	size_t param_count;
+	int err;
+
+	/* The peripheral does not accept any ISO requests if security is
+	 * insufficient, so we only need to handle central here.
+	 * BT_ISO_ENCRYPT_PENDING is only set by the central.
+	 */
+	if (!IS_ENABLED(CONFIG_BT_CENTRAL) ||
+	    acl->role != BT_CONN_ROLE_CENTRAL) {
+		return;
+	}
+
+	param_count = 0;
+	for (size_t i = 0; i < ARRAY_SIZE(iso_conns); i++) {
+		struct bt_conn *iso = &iso_conns[i];
+		struct bt_iso_chan *iso_chan;
+
+		if (iso == NULL || iso->iso.acl != acl) {
+			continue;
+		}
+
+		iso_chan = iso_chan(iso);
+		if (iso_chan->state != BT_ISO_ENCRYPT_PENDING) {
+			continue;
+		}
+
+		bt_iso_chan_set_state(iso_chan, BT_ISO_DISCONNECTED);
+
+		if (hci_status == BT_HCI_ERR_SUCCESS) {
+			param[param_count].acl = acl;
+			param[param_count].iso_chan = iso_chan;
+			param_count++;
+		} else {
+			BT_DBG("Failed to encrypt ACL %p for ISO %p: %u",
+			       acl, iso, hci_status);
+
+			/* We utilize the disconnected callback to make the
+			 * upper layers aware of the error
+			 */
+			if (iso_chan->ops->disconnected) {
+				iso_chan->ops->disconnected(iso_chan,
+							    hci_status);
+			}
+		}
+	}
+
+	if (param_count == 0) {
+		/* Nothing to do; this shouldn't happen */
```

Comment:
If this shouldn't happen, should there be a BT_WRN or BT_ERR here?

Reply:
Added a `BT_ERR`

Reply:
Remove the `BT_ERR` again, and update the comment. 

This can actually happen if no CIS are pending encryption, but the ACL is encrypted by other means.